apps/antalmanac/src/components/buttons/LoadSchedule.tsx

```diff
@@ -0,0 +1,140 @@
+import { useCallback, useEffect, useState } from 'react';
+import {
+    Button,
+    type ButtonProps,
+    Checkbox,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    FormControlLabel,
+    TextField,
+    Tooltip,
+} from '@mui/material';
+import { LoadingButton } from '@mui/lab';
+import { CloudDownload as CloudDownloadIcon } from '@mui/icons-material';
+
+import { loadSchedule } from '$actions/AppStoreActions';
+import { isDarkMode } from '$lib/helpers';
+import { useLoadingStore } from '$stores/loading';
+
+function getSavedUserId(): string | null {
+    if (typeof window === 'undefined') {
+        return null;
+    }
+
+    const userId = window.localStorage.getItem('userID');
+    return userId;
+}
+
+export function LoadScheduleButton(props: ButtonProps = {}) {
+    const [loadedSchedule, setLoadedSchedule] = useLoadingStore((state) => [
+        state.loadedSchedule,
+        state.setLoadedSchedule,
+    ]);
+
+    const [open, setOpen] = useState(false);
+
+    const [rememberMe, setRememberMe] = useState(true);
+
+    const [loading, setLoading] = useState(false);
+
+    const [userId, setUserId] = useState(getSavedUserId() ?? '');
+
+    const handleUserIdChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+        setUserId(event.target.value);
+    }, []);
+
+    const handleRememberMeChange = useCallback((_event: unknown, checked: boolean) => {
+        setRememberMe(checked);
+    }, []);
+
+    const handleOpen = useCallback(() => {
+        setOpen(true);
+    }, []);
+
+    const handleClose = useCallback(() => {
+        setOpen(false);
+    }, []);
+
+    const handleLoadSchedule = useCallback(async (userId: string, rememberMe: boolean) => {
+        setLoading(true);
+        await loadSchedule(userId, rememberMe);
+        setLoading(false);
+    }, []);
+
+    const handleSubmit = useCallback(() => {
+        handleLoadSchedule(userId, rememberMe).then(() => {
+            setOpen(false);
+        });
+    }, [userId, rememberMe, handleLoadSchedule]);
+
+    /**
+     * Try to find the username stored locally and automatically load their schedule.
+     *
+     * Only happens once when opening the application for the first time.
+     */
+    useEffect(() => {
+        // If already attempted to load the schedule for the first time, don't try again.
+        if (loadedSchedule) {
+            return;
+        }
+
+        const userId = getSavedUserId();
+
+        if (userId !== null) {
+            handleLoadSchedule(userId, rememberMe);
+        }
+
+        setLoadedSchedule(true);
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [loadedSchedule]);
```

Comment:
eslint wants us to include all stateful variables, i.e. `userId, rememberMe`, but we don't really want `useEffect` to run in those cases, since only the first app load matters. Technically we could include the dependencies and run it, and it would be expected to exit immediately (i.e. after the first type guard, basically acting like a noop)